### PR TITLE
GO-2579 Set LastUsedDate for initial types

### PR DESCRIPTION
**Pull request by KirillSto**
---

- [x] I understand that contributing to this repository will require me to agree with the [CLA](https://github.com/anyproto/.github/blob/main/docs/CLA.md)

---

### Description

This PR sets lastUsedDate in a way that Note-Page-Task-Set-Collection appear first in sort on new spaces

### What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI

### Related Tickets & Documents

https://linear.app/anytype/issue/GO-2579/set-initial-state-in-lastuseddate-relation-for-note-page-task-set

### Added tests?

- [x] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 [tech-docs](https://github.com/anyproto/tech-docs)
- [x] 🙅 no documentation needed

### CODE REVIEW

**Review by AnastasiaShemyakinskaya — APPROVED** [review]

**Review by AnastasiaShemyakinskaya — APPROVED** [review]

File: core/block/object/objectcreator/creator.go [hunk 1]
```diff
@@ -80,109 +69,40 @@ func (s *service) Init(a *app.App) (err error) {
...
+// CreateObject is high-level method for creating new objects
 func (s *service) CreateObject(ctx context.Context, spaceID string, req CreateObjectRequest) (id string, details *types.Struct, err error) {
 	space, err := s.spaceService.Get(ctx, spaceID)
 	if err != nil {
 		return "", nil, fmt.Errorf("get space: %w", err)
 	}
-	return s.CreateObjectInSpace(ctx, space, req)
+
+	if req.ObjectTypeKey.String() == "" {
+		objectTypeKey, err := domain.GetTypeKeyFromRawUniqueKey(req.UniqueKey)
```
Comment (deff7) [inline core/block/object/objectcreator/creator.go:93]: what's the purpose of this? 

CreateObjectUsingObjectUniqueTypeKey was pretty specific method

File: core/block/editor/objecttype/lastused.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+		log.Errorf("failed to set lastUsedDate to type object '%s': %w", key.String(), err)
+	}
+}
+
+func SetLastUsedDateForInitialObjectType(id string, details *types.Struct) {
+	if !strings.HasPrefix(id, addr.BundledObjectTypeURLPrefix) || details == nil || details.Fields == nil {
+		return
+	}
+
+	var decrement time.Duration
```
Comment (deff7) [inline core/block/editor/objecttype/lastused.go:52]: maybe use `priority`? 

```
switch id {
	case bundle.TypeKeyNote.BundledURL():
		priority = 1
	case bundle.TypeKeyPage.BundledURL():
		priority = 2
```

File: core/block/object/objectcreator/smartblock.go [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+		SmartblockType: sbType,
+		InitFunc:       initFunc,
+	})
+}
+
+func generateRelationKeysFromState(st *state.State) (relationKeys []string) {
+	if st == nil {
+		return
+	}
+	details := st.CombinedDetails().GetFields()
```
Comment (deff7) [inline core/block/object/objectcreator/smartblock.go:144]: CombinedDetails makes unnecessary allocations. Just iterate details and local details separately

File: core/block/object/objectcreator/creator.go (hunk 1 above)
Comment (KirillSto) [reply core/block/object/objectcreator/creator.go:93]: We are including specific fields of CreateObject request in CreateObjectRequest structure. We can make specific method for each field: CreateObjectUsingTemplate or CreateObjectWithDetails. Maybe we should leave it more general and use single CreateObject method in all places?

File: core/block/editor/objecttype/lastused.go (hunk 2 above)
Comment (KirillSto) [reply core/block/editor/objecttype/lastused.go]: Done!

File: core/block/object/objectcreator/installer.go [hunk 4]
```diff
@@ -192,6 +200,10 @@ func (s *service) prepareDetailsForInstallingObject(ctx context.Context, sourceS
 	details.Fields[bundle.RelationKeySourceObject.String()] = pbtypes.String(sourceId)
 	details.Fields[bundle.RelationKeyIsReadonly.String()] = pbtypes.Bool(false)
 
+	if isNewSpace {
```
[the remark this answers is not in the document]
Comment (KirillSto) [reply core/block/object/objectcreator/installer.go:203]: There was a mistake. IsNewSpace should be equal false on import. Fixed!

File: core/block/object/objectcreator/smartblock.go (hunk 3 above)
Comment (KirillSto) [reply core/block/object/objectcreator/smartblock.go]: Done!

**Pull request merged**